typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/require-await': 'error'`

Reply:
This rule is disabled in the extensions configuration.
**Proposal**
Leave this rule un-configured.

Reply:
The config is already arranged so that the extensions takes priority over the recommended rule, so we don't need to do anything here.